tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1303 @@
+      let msg2 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await otherNode.publish(pubsubTopic, msg2)
+
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg2) == handlerFuture.read()
+        (pubsubTopic, msg2) == otherHandlerFuture.read()
+
+      # Given node is stopped and restarted
+      await allFutures(switch.stop(), node.stop())
+      await allFutures(switch.start(), node.start())
+
+      # When sending a message from node
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      let msg3 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await node.publish(pubsubTopic, msg3)
+      
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg3) == handlerFuture.read()
+        (pubsubTopic, msg3) == otherHandlerFuture.read()
+
+      # When sending a message from other node
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      let msg4 = fakeWakuMessage(testMessage, pubsubTopic)
+      discard await otherNode.publish(pubsubTopic, msg4)
+      
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg4) == handlerFuture.read()
+        (pubsubTopic, msg4) == otherHandlerFuture.read()
+
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    xasyncTest "Relay can receive messages after subscribing and stopping without unsubscribing":
+      # Given a second node connected to the first one
+      let
+        otherSwitch = newTestSwitch()
+        otherPeerManager = PeerManager.new(otherSwitch)
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let 
+        otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+        otherPeerId = otherRemotePeerInfo.peerId
+
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      # Given both are subscribed to the same pubsub topic
+      var otherHandlerFuture = newPushHandlerFuture()
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherHandlerFuture.complete((topic, message))
+
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      check:
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.subscribedTopics == pubsubTopicSeq
+      
+      await sleepAsync(500.millis)
+
+      # FIXME: Inconsistent behaviour with Filter protocol.
```

Comment:
Filter is able to receive messages even if it has been stopped.

Reply:
How can messages be received and sent with the node and the switch stopped? I'm missing probably missing something :)

Reply:
Here you go, [issue comment](https://github.com/waku-org/nwaku/issues/2178#issuecomment-1791338580) that answers your question.